ui/uppy/src/plugins/Dashboard.jsx

```diff
@@ -25,580 +25,593 @@ import UppyDashboard from '@uppy/dashboard';
-    plugins.forEach((pluginID) => {
-      const plugin = this.uppy.getPlugin(pluginID);
-      if (plugin) {
-        plugin.mount(this, plugin);
-      }
-    });
-
-    if (!this.opts.disableStatusBar) {
-      this.uppy.use(StatusBar, {
-        id: `${this.id}:StatusBar`,
-        target: this,
-        hideUploadButton: this.opts.hideUploadButton,
-        hideRetryButton: this.opts.hideRetryButton,
-        hidePauseResumeButton: this.opts.hidePauseResumeButton,
-        hideCancelButton: this.opts.hideCancelButton,
-        showProgressDetails: this.opts.showProgressDetails,
-        hideAfterFinish: this.opts.hideProgressAfterFinish,
-        locale: this.opts.locale,
-        doneButtonHandler: this.opts.doneButtonHandler
-      });
-    }
-
-    if (!this.opts.disableInformer) {
-      this.uppy.use(Informer, {
-        id: `${this.id}:Informer`,
-        target: this
-      });
-    }
-
-    if (!this.opts.disableThumbnailGenerator) {
-      this.uppy.use(ThumbnailGenerator, {
-        id: `${this.id}:ThumbnailGenerator`,
-        thumbnailWidth: this.opts.thumbnailWidth,
-        thumbnailType: this.opts.thumbnailType,
-        waitForThumbnailsBeforeUpload: this.opts.waitForThumbnailsBeforeUpload,
-        // If we don't block on thumbnails, we can lazily generate them
-        lazy: !this.opts.waitForThumbnailsBeforeUpload
-      });
-    }
-
-    // Dark Mode / theme
-    this.darkModeMediaQuery =
-      typeof window !== 'undefined' && window.matchMedia ? window.matchMedia('(prefers-color-scheme: dark)') : null;
-
-    const isDarkModeOnFromTheStart = this.darkModeMediaQuery ? this.darkModeMediaQuery.matches : false;
-    this.uppy.log(`[Dashboard] Dark mode is ${isDarkModeOnFromTheStart ? 'on' : 'off'}`);
-    this.setDarkModeCapability(isDarkModeOnFromTheStart);
-
-    if (this.opts.theme === 'auto') {
-      this.darkModeMediaQuery.addListener(this.handleSystemDarkModeChange);
-    }
-
-    this.discoverProviderPlugins();
-    this.initEvents();
-  };
+	initEvents = () => {
+		// Modal open button
+		if (this.opts.trigger && !this.opts.inline) {
+			const showModalTrigger = findAllDOMElements(this.opts.trigger);
+			if (showModalTrigger) {
+				showModalTrigger.forEach((trigger) => trigger.addEventListener('click', this.openModal));
+			} else {
+				this.uppy.log(
+					'Dashboard modal trigger not found. Make sure `trigger` is set in Dashboard options, unless you are planning to call `dashboard.openModal()` method yourself',
+					'warning'
+				);
+			}
+		}
+
+		this.startListeningToResize();
+		document.addEventListener('paste', this.handlePasteOnBody);
+
+		this.uppy.on('plugin-remove', this.removeTarget);
+		this.uppy.on('file-added', this.hideAllPanels);
+		this.uppy.on('dashboard:modal-closed', this.hideAllPanels);
+		this.uppy.on('file-editor:complete', this.hideAllPanels);
+		this.uppy.on('complete', this.handleComplete);
+    this.uppy.on('upload-success', this.checkInProgressFiles);
```

Comment:
Added an extra verification when files are uploaded (or when they fail) to update the `In Progress` state.